### PR TITLE
Add XML context, Since XML and JSON share the Response, new concept: ResponseStorage.

### PR DESCRIPTION
**Pull request by stephpy**
- [x] Add tests

### CODE REVIEW

File: src/Json/JsonContext.php [hunk 1]
```diff
@@ -5,8 +5,9 @@
 use mageekguy\atoum\asserter\generator as asserter;
 use Behat\Behat\Context\BehatContext;
 use Behat\Gherkin\Node\PyStringNode;
+use Rezzza\RestApiBehatExtension\Response\ResponseStorageAware;
 
-class JsonContext extends BehatContext implements JsonStorageAware
+class JsonContext extends BehatContext implements ResponseStorageAware
```
Comment (shouze) [inline src/Json/JsonContext.php]: JCVD?™

Comment (stephpy) [inline src/Json/JsonContext.php]: ![](data:image/jpeg;base64,/9j/4AAQSkZJRgABAQAAAQABAAD/2wCEAAkGBxMSEBUSEhIVFRIVFxQQEhAWEBQUEBAUFBIWFhQUFBQYHCggGBolHBQUITEhJSkrLi4uFx8zODMsNygtLisBCgoKDg0OFxAQGCwfHBwsLCwsLCwsLCwsLCwsLCwrNywsLCwsLCwsLCw3LCssKzcsNzc3KyssLCwrLCwrLCssK//AABEIAMABAAMBIgACEQEDEQH/xAAbAAABBQEBAAAAAAAAAAAAAAADAAECBAUGB//EADUQAAIBAwIEBAUDAwQDAAAAAAABAgMEESExBRJBURMiYXEGMlKBkRShsULB0QcjcvBikvH/xAAZAQACAwEAAAAAAAAAAAAAAAACAwABBAX/xAAiEQADAAICAgMBAQEAAAAAAAAAAQIDESExBBIiQVETMkL/2gAMAwEAAhEDEQA/APNB0js/gDhtvWqyVblxy5Sbwn+TC+KbanTupxpPME9BEZFXQ/Jic/ZkYEyRNDRakCOaEbdEa9BJZQO0E8bRSZHBpULdSQO5pRjp1L2glhZTUcko030WTTtrSnHllVb5WubkW+HtqP8Ar4xSVOCwk1lrLf3A939INYEuaoqUbKpPHLBvL5VhPV9F+zAOnv3WjS3TNGnxurHDUsNYS0XRYXT1YGzvvDm5xWJPfGi/BN12wnGP6ZVhOUdv4Ojq/FCq28aM6MIyjhqtDRv/AJQx/cyYXFN/NHVttvK6/YjVoQalKL215dFp9wfWafJe6lcPgpSjq8bNsbwiTeuBmN6MzXI04hIxRc4ZwWvcvFKm591lL+RcU4HXtny1qbg99WmmvRonsiOWvoqZH50AY3KQpMJKZByFgj1ICLI4sCIT2CJ6EcDInAF1pbGTKoE49RcrPU/gbglnXpTdWnCUsrfGVHG6Z55xqlCFzUVJ5p8zUfZMGcs10G8Hr9mfgZIu7gpwGJiqxtABmtQjiRaLBSewrqOOz/cHOWR+X1LfCuHSr1Y047yeNdgF6oZMVRTprOxKUGlroad9wuVtV8Oo1tnK2fsVrqvlcu+OpXsN/nrstQksaArp+Us0YrC9gHEasUuVbgrbDp6RC3qqNJ9yk3ncbm/+EcjEtGd5GSlUb3G5xhYL0gW2xZHQlFvoT8CXZ/gm0Epf4QkJMXI1un+BNE4Be0IeMhsDlNbRSb2dr/p9xqNCulNpRlmLYf8A1N47TrVIQpSUlGPmaWmX6nFJeUDKWohYmq2PV74GUCbQ9OLYaVB8uR2w3hetgNCPKSTE2EhXwG5R0kDnIaTIT2n8COKHUECkxE+tE919ItUrmUVhSa+4KWrB5LnDoJ5yKepNGDH/AFAqIzNSVBYKta2xnCJN7G5/GcIoSG3JSg87ClSaG7Of6v8ABSjhh7e7lTkpQeJLZgq3zexCW5Xqi/6NdFm+vZ1p883mXcBHci3gUWTSLm6bNqKah64MRPq+pvU9kZt9b4fMtnuuwMhZJfZTkKMM7ISNiycacc4y2XVaDw4prsFZ8KlLDeiNalwemlrHPuwdHibb1joatC4UlojNV0dbFgxpdFelbwi8KKX2JVIr/qLTTyZ1/dtPCQK2xlVMroa4pJ9F+DJubNahG6r6gpuot9fsNngw5nNf8mfOGHgjjUsVHncDI0JnOpE4LOge3sG9WgNGWNTetZNoTlv1NfjTO9gKNm1siVzRag2zRjkrcST5GJi22afIv46OZItD5ItmxI49MaTGRNjIspESUUSpJDzZRYPBr8FopptmTHU3eAw8n3M2Y3+Iy7OlHGwJ0o42LkqYN0RMM13tozadJOrLToHnTQreK8SQdw0GTT2K9eDnL1f7jAssXq/3GVzYjl6GHQsCbBokdm3ReYojdLyv2HtvkRK4j5WLRqfJmQop0ufs9TcsreKjllOwoqVGUe+pcjFunBe2WDbNeCEl0WpTg3jHsGoy5WF4ZaxVBuppUb2WNI9BvBjrqxJvkNd1ko5MeVVMNxCTcfRDWtDmp5W+dfZFoXe2+AcajfyrK2wlqQlVTeGmvdF3hsPDlKTnp0WNQd7JVHnqHwhVOn2gdtw+E5a7PRlPi/DOWSiuuxo8OfLNeuhZ49Raw19hk0Y8sI42OjcXujoLCXlRkVqXNP8Anua1u102xjs19hObkLxpaRfiyvxLWmx4zB38v9tgYlyFnekctF6khkhZNyOWxMQhFlCyLIiLIUTTN7gmOR+5gKRucFfkfuZsx0fGNVyIORFzB8xnmdGunwV7Z+eRYnPQqW/zyC1ZaDoXIlsw6zzN/wCRU7eU3yxWX2FKm5VMRWW9Mep2vC7GNGGGk5byf9sjHmSEYfGeQ4qdrOHzRwQjHD2O34pZRa5itZUY+FnlWctPQizbGPwmn2ZNDZewO9+V51NShTi6ijjR5+wO/slF9WsrKK9xi8atk+GW/JQi3/VqljYuUqWYoDXv4yWI9NEgtCppqItvZ0ccSlosU6a7D1orZDU5ruAq8QjGphtZ6IqRj0kVb6n2G4XWaWH3wyPFuIRTT2KsKuP9xbdV3DcmdUt9m1OS/JVmgkaicUwVSoig6SFRjma9zf45WiqEl1UdPfBy067T0I179zpSy3nb8jE9IxWtsDwKjmo090s+2WbMbBZb1w9SpwK3UYc76vGeuDYytYsTT2aseNaKtS0T+XfcoXdtLkaW5pWkuWpJPto/7B66yti4emLzYFSOCuKEovzLG+H0AnbOzjUUoPt+H0aOLqQak1LeLaZtl7RxcuNy9E6J0nw58Ju6i6jk4xzhYWrOXTO5+GfiqnRoxg1sY/JyZJ/ybPEjE18uyr8R/BLtqbqxm5JbprXByVN5R6Zxf4ojWoygk9Uedxs5rp7DcV05+QvNiSr4oFyGzw75DO/Tz7GlZxxHUq+WNwrXZYbITY7n2ByYKGMDa9fcJUloVrZ4z7hJy0YcrkTTNTgtkqa8SaxOXpsi7cOLWjLM9VqYt1LlegnR100ujQrPmpqOdtW/Qayio0Fnq2ypBtrXYs31ZcmnsgkgaekZc7lRrQedMvJe4jVjjfJz9WOZa66r+Toq9tGUNvRfgPQtttcGHY0nKfovMzV5/PIbhFBRjLO+dfYVaa8TRrbzLtqVcg4ae+S3RlkhWpRctskYzBVbqMHqDPBrprQ0uHR3evuCr0YvC2S6JBpcSj6/gBUuIv0CaZn3KYTmSWFsgUpgZTysgHN5+5SQN3sLWK1usyiujeQ0nkVtDzZ7B64ES/kb1tFeEsPZILlSjlbop2NwlDKTw86b4x3IWV03zYi+VvMX/YS+zd9cDXd5GdWEY/NzLK7dzfqQzkweD2+ZyqyWG9IrtjdnQqOhN8kSbRky8k00c98U0eS45l8s0pL3W50t5DPoc/8AEVeM4wSeZQbz7GnHs5PlStmJJdhW8MyRY8TpgJZQy9S7aM+KOTShPCHTIxJCTaiSmNUq4TIyKl9LEQpkF0Qsq2rTLbkZdJal5z0+xbkF2Rttn7hKr0BWr8oq8tAoXIumdfNGLeU9TbrIoV4ZaRm2dipYKCXKkx3SyWaVr1C+HgnsRL9MS54XnVSwbDp4innK79cEa8kUq11ywcc7fwHPLFZGolmZe37hJ8ujaxn+5T4fXxPfOd2ytdVeaTAqWHlD3HBy8efVnVU5EHZRlq39jPsa+Y+vVGgqbl1Evg6q1a32G8KKxlL3KtZQT2LUeHJ76sDWskiJlNP6RRi12BzeGWJ00njJUrx10DSMd1oUqgelLH7lRIVSrhYyHoz/ANeTR+HLl+el0fmXp0Z0FOjjc5TgVflrej0OuhU6GTLvZ1/HpOeQlvhBbm7jThzSeEt2Vak0tTj+McQdab1agniKzp7+4zFjb5EeT5f81pFnifHJTliGVDbPWXqY79/2GQmjYkkcW7dvbYajW5cbP0Zco8Qx0RmIkmC4Rc5ak3Y30cZFG9h1MWMWScBTxaNM+Q2atS8h2KVeqpPRMHHKCxrPsia0HvfYOMH2LdK3k0QV4+yC077/AMSm2XqSULVpbkK9B4epN3y7CqXScQofIutHTXFTQr0+5KpUyV69VJGfR2HekaSqJIrVa5l1L5LqU63EtAljZnvyUkW7y9Ud2Y1ze8zePYFfNYWJc0nrLGy9CsaohI5WbyKvgZiyOMw2tmXegttVcZJnQUK+NcnNwRvWWJw1M1o6viZXo1aV2l1Kt9dR7lKvQlumZdxKfUkQHm8prjRbq1nnRgZVytKTGyOUmB5XTC+ICkxyEy2hVMUZvOfwblnxXK1ev8mESQusaYyPKuejd4jxXyOK3ksZ7GAiWokMlaWgLyO3yLAyHchFgEWxc7EMyyBKdUKporJIl7AtFqtFzmHKUWwkauAND5y/paEDhVywrQI7aYkiFUmQq7Bz2LpIv0+LqT1BXF9phGSthczK/kgK8uqQapWb3B8xHJEZozOmx2yOSTYsEIM2JISRNLJCaImrwmejXqZkoliwq4eH1E5J2jb4t6rRvZ0KlWnksUtiNWOolPR0blV9GZUsuxB2MsGktwyjoNWQzPx5XJhSoOO4KcMGleVEjOnUyORgypJ8EELI5GSLEkkNzCQ7RCewmxmJschNjCHbIshBRRLJEchByLHGZNFMkpliMyoEhIBpDZtllVEQqT0GkDmi1wMdA2siGTHbCMwhDZHSIQYdMflH5SFkWNElgZkIT5tBl3IRYQprYSty00aVrfxxiX5LX62DXzGC2STFPEma15tfhuSqxX9S/Opn3l63otik0MgpxpCsnkXY85tkUh8Cewxmb7EPGDlsNDV4S30NBtUvKsSltKSw4/b1BbHxjT7AO2cY8z+y9yy7FOOn3WucglPmby3+dP4GhWecJ7bALex7mEgFeg4PDxnGVr0BJlu8q+Jyyer1Tb1b92VWhqMVLkQkhiUSEI5EMO0WltlNixklBxT82y3wGtqXiSUFiPd9/sDp2y5pRctYvCxHLl++hqnxm10Z3mS7Y1Omm98JZfuDyXp8OwpOU9FKMNEm8y7rJXu6XK3TeMxeXLuKzYXHaDjKq6ZGMhSWSGB1LAgfL2CRJEWiUQgRDNEsCKLHExJkiFkR0MxkQgmJIdCIQSJZIEkyEGeR5IcTZCmRGbEPFalMuOQtJ4zjGunqidOGf8C8PZfkKln+z7AbNMpj1P6Uuu4OvHG3vj/uwT108mjTWFr1Yq6zPRad8boiZd7KnORbHj1IsYjOxCFkRARxsiyMwprTBaLllzOtDPLFyflnNYin3yjZXCJLnfjW2ZPEszaz6o53PM0pSeO7y8D06fNLli+6y9Njp4/IlSZLxPZu0+FSzOmq1Dlj5/meJNbcpiTrTfNNr5vK21pp2YKnFdXj7LUTm2vTt0MvkZvcbjnX0NgZ5yOhmYzRL0IcZ4W8l+3+SMqsF/UshBemlyETECnXh9SH/UQ25l+dSitE0h8g/Hhj51/H9xnXj9aITQRsZshGtHv+w8pLuvwyygkRJEPGh9SI/qIfUvyUQI9ySkCdeH1oZ3EfqRCBmQkyLuIfUvyN40PrRCBYsLRj5kVvGgv6kEjcU9H4n7FDcei1UqOLWNNPsyzQXNJNPTthaFR3NNp+eONHrPlefTcr/qIpZU469FLIFQ30aFkUv5dHSOzSilOSSlpGLfzf8U9jGrwUZySf4enuV/16xrU193/kHKpDPzx/9iolrsrJkmv8oJkgyH6iH1L8jePD60NMbZMcE68fqX5HVxD6l+SFBBIF48fqQvHj9SIWFYxDx4/UhePD6kWU0EbGB/qIfUvyLx4fWvyUUTyOmC8eH1oi7iPdfksOT//Z)

**Comment by shouze** [comment]
:+1:

**Comment by tyx** [comment]
:+1:

**Pull request merged**